### ISSUE TITLE
Not working on High Sierra GM Build

### ISSUE THREAD

**Issue by iamsankalp** [romannurik/Sketch-ArtboardTricks#18]
Hi, Roman! Thanks for the awesome plugin.

As soon as I try to run the plugin to arrange artboards - Sketch crashes. I had installed it via Sketch Runner.

Is this a known issue?

**Comment by romannurik** [comment]
I don't have High Sierra yet so unfortunately can't troubleshoot :-/ seeing the log from Console.app would be useful!

**Comment by olafchou** [comment]
what should i do ??   could tell me which step should i operate ?

**Comment by iamsankalp** [comment]
Hi, Roman!

Uploading an archive with two files.

1) sketch_console - this is from the console.app. I started console app, then started sketch and run the plugin once. After which it crashed. Then searched for 'Sketch' in console and put up this log in text file.

2) sketch_quit_details - this if from 'Sketch quit unexpectedly' native dialog details.

Hope this is of help.

Sankalp

[sketch_log.zip](https://github.com/romannurik/Sketch-ArtboardTricks/files/1346692/sketch_log.zip)

**Comment by olafchou** [comment]
[sketch_log.zip](https://github.com/romannurik/Sketch-ArtboardTricks/files/1346855/sketch_log.zip)

**Comment by romannurik** [comment]
Thanks all! @iamsankalp in your log, it *looks* like the plugin command is exposing some sort of bug in Sketch, but it's not clear what code in the plugin is triggering this. I'll have to try to get my hands on High Sierra and troubleshoot. Don't know of any workarounds in the meantime, sadly :-/

FWIW, this is the closest thing to a culprit I've found:

```
error	00:20:48.831082 +0530	Sketch	Unable to simultaneously satisfy constraints:
(
    "<NSAutoresizingMaskLayoutConstraint:0x608000686db0 h=--& v=--& MSGradientView:0x604000580dd0.width == 0   (active)>",
    "<NSAutoresizingMaskLayoutConstraint:0x60800009bd50 h=-&- v=&-- H:[NSTextField:0x7ff4c4775460]-(35)-|   (active, names: '|':MSGradientView:0x604000580dd0 )>",
    "<NSAutoresizingMaskLayoutConstraint:0x60800048ad70 h=-&- v=&-- NSTextField:0x7ff4c4775460.minX == 7   (active, names: '|':MSGradientView:0x604000580dd0 )>"
)

Will attempt to recover by breaking constraint 
<NSAutoresizingMaskLayoutConstraint:0x60800009bd50 h=-&- v=&-- H:[NSTextField:0x7ff4c4775460]-(35)-|   (active, names: '|':MSGradientView:0x604000580dd0 )>

Set the NSUserDefault NSConstraintBasedLayoutVisualizeMutuallyExclusiveConstraints to YES to have -[NSWindow visualizeConstraints:] automatically called when this happens.  And/or, set a symbolic breakpoint on LAYOUT_CONSTRAINTS_NOT_SATISFIABLE to catch this in the debugger.
default	00:20:48.955666 +0530	systemsoundserverd	686: -> Incoming Request : actionID 4096, inClientPID 4046(Sketch), inBehavior 1, customVibeDataProvided 0, loop 0, loopPeriod 0.000000, inFlags 0, inClientCompletionToken 2
default	00:20:51.101544 +0530	Sketch	LSExceptions shared instance invalidated for timeout.
default	00:20:51.655451 +0530	Sketch	0x7ff4c4457a60 - PerformanceMonitor::measurePostBackgroundingMemoryUsage: Process was using 220209152 bytes of memory after becoming non visible.
default	00:20:52.944325 +0530	kernel	AGC:: [Sketch pid 4046 mux-aware] exiting, non-mux-aware app count 0, runtime: 0:00:11.535
```

**Comment by ugaldew** [comment]
I wil be looking forward, your plug in is incredibly helpful ... Thank you Roman

**Comment by patrickplaggenborg** [comment]
Broken here as well.

Just updated both to High Sierra and Sketch 47. Thought it was Sketch, but is probably related to this issue report about High Sierra.

Submitted a crash report to Sketch and to Apple, if that helps. Hope it can be fixed in the plugin itself.

Glad you're on it, thank you!

**Comment by romannurik** [comment]
Hey folks, can someone try this build?

[ArtboardTricks.zip](https://github.com/romannurik/Sketch-ArtboardTricks/files/1396377/ArtboardTricks.zip)

**Comment by herkulano** [comment]
@romannurik works! 💯

**Comment by ugaldew** [comment]
It ... WORKS!!! Thank you Roman ! this plug in is pure genius !

**Comment by jasmussen** [comment]
Can confirm it works for me too. High Sierra and Sketch v. 47.1 (45422), latest as of writing this. Thank you!

**Comment by iamsankalp** [comment]
Works perfect. I have tested it on High Sierra Beta 13.1 with Sketch 47.1.

Thanks, @romannurik! Really appreciate your time.

**Comment by patrickplaggenborg** [comment]
Replaced my current 1.0.5 with this build and works for me as well!

Thanks @romannurik!

(High Sierra 10.13 (17A405) and Sketch 47.1 (45422).

**Comment by romannurik** [comment]
Great! Will push 1.0.6 with the fix today.

**Comment by romannurik** [comment]
Fixed in 1.0.6
